### PR TITLE
Contracts/feature/collateral commitment ratio

### PR DESCRIPTION
**Pull request by ethereumdegen**

### CODE REVIEW

File: packages/contracts/contracts/LenderCommitmentForwarder.sol [hunk 1]
```diff
@@ -363,30 +362,41 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
...
+                _maxPrincipalPerCollateralAmount,
+                MathUpgradeable.Rounding.Up
+            );
+        /*
         return
             MathUpgradeable.ceilDiv(
                 _principalAmount,
                 _maxPrincipalPerCollateralAmount
-            );
+            );*/
```
Comment (passabilities) [inline packages/contracts/contracts/LenderCommitmentForwarder.sol]: delete

File: packages/contracts/contracts/LenderCommitmentForwarder.sol [hunk 2]
```diff
@@ -363,30 +362,41 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
 
     function getRequiredCollateral(
         uint256 _principalAmount,
-        uint256 _maxPrincipalPerCollateralAmount,
+        uint256 _maxPrincipalPerCollateralAmount, //should be expressed expanded by collateralTokenDecimals and principalTokenDecimals
```
Comment (passabilities) [inline packages/contracts/contracts/LenderCommitmentForwarder.sol:365]: lets add a comment block here for the function

File: packages/contracts/contracts/LenderCommitmentForwarder.sol [hunk 3]
```diff
@@ -363,30 +362,41 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
 
     function getRequiredCollateral(
```
Comment (passabilities) [inline packages/contracts/contracts/LenderCommitmentForwarder.sol:359]: lets add a comment block for the function

File: packages/contracts/contracts/LenderCommitmentForwarder.sol (hunk 2 above)
Comment (passabilities) [inline packages/contracts/contracts/LenderCommitmentForwarder.sol]:
```suggestion
        uint256 _maxPrincipalPerCollateralAmount, //should be expressed expanded by (principalTokenDecimals * 2)
```

File: packages/contracts/contracts/LenderCommitmentForwarder.sol [hunk 4]
```diff
@@ -363,30 +362,41 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
...
+        address _collateralTokenAddress,
+        address _principalTokenAddress
     ) public view virtual returns (uint256) {
         if (_collateralTokenType == CommitmentCollateralType.NONE) {
             return 0;
         }
+
+        uint8 collateralDecimals = 0;
+        uint8 principalDecimals = 0;
+
```
Comment (passabilities) [inline packages/contracts/contracts/LenderCommitmentForwarder.sol]:
```suggestion
        //principal will always be ERC20 
        uint8 principalDecimals = IERC20MetadataUpgradeable(_principalTokenAddress)
            .decimals();
        uint8 collateralDecimals;
```

File: packages/contracts/contracts/LenderCommitmentForwarder.sol [hunk 5]
```diff
@@ -363,30 +362,41 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
...
-                    _principalAmount,
-                    10**decimals, //multiply by the collateral token decimals 
-                    _maxPrincipalPerCollateralAmount,
-                    MathUpgradeable.Rounding.Up
-                );
         }
+
+        //principal will always be ERC20 
+        principalDecimals = IERC20MetadataUpgradeable(_principalTokenAddress)
+            .decimals();
```
Comment (passabilities) [inline packages/contracts/contracts/LenderCommitmentForwarder.sol]:
```suggestion
```

File: packages/contracts/contracts/LenderCommitmentForwarder.sol [hunk 6]
```diff
@@ -363,30 +362,41 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
...
+    
+
+        return
+            MathUpgradeable.mulDiv(
+                _principalAmount, 
+                (10**(collateralDecimals + principalDecimals) ), //multiply by the collateral token decimals
```
Comment (passabilities) [inline packages/contracts/contracts/LenderCommitmentForwarder.sol]: fix comment

**Review by passabilities — CHANGES_REQUESTED** [review]
please fix all comments and function name for each test to match what is actually being tested

File: packages/contracts/contracts/LenderCommitmentForwarder.sol (hunk 3 above)
Comment (ethereumdegen) [reply packages/contracts/contracts/LenderCommitmentForwarder.sol:359]: done

File: packages/contracts/contracts/LenderCommitmentForwarder.sol (hunk 2 above)
Comment (ethereumdegen) [reply packages/contracts/contracts/LenderCommitmentForwarder.sol]: not true

File: packages/contracts/contracts/LenderCommitmentForwarder.sol (hunk 4 above)
Comment (ethereumdegen) [reply packages/contracts/contracts/LenderCommitmentForwarder.sol:384]: need to use an if statement around the IERC20 call here still . Dont think we need to change anything.

File: packages/contracts/contracts/LenderCommitmentForwarder.sol (hunk 5 above)
Comment (ethereumdegen) [reply packages/contracts/contracts/LenderCommitmentForwarder.sol:392]: Why change this

File: packages/contracts/contracts/LenderCommitmentForwarder.sol (hunk 6 above)
Comment (ethereumdegen) [reply packages/contracts/contracts/LenderCommitmentForwarder.sol]: done

File: packages/contracts/contracts/LenderCommitmentForwarder.sol (hunk 1 above)
Comment (ethereumdegen) [reply packages/contracts/contracts/LenderCommitmentForwarder.sol]: done

File: packages/contracts/contracts/LenderCommitmentForwarder.sol [hunk 7]
```diff
@@ -361,31 +348,52 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
...
-                    MathUpgradeable.Rounding.Up
-                );
+            collateralDecimals = IERC20MetadataUpgradeable(
+                _collateralTokenAddress
+            ).decimals();
         }
+
+        //principal will always be ERC20
+        principalDecimals = IERC20MetadataUpgradeable(_principalTokenAddress)
+            .decimals();
```
Comment (passabilities) [inline packages/contracts/contracts/LenderCommitmentForwarder.sol]: duplicated

```suggestion
```

File: packages/contracts/contracts/tests/LenderCommitmentForwarder_Test.sol [hunk 8]
```diff
@@ -488,24 +526,121 @@ contract LenderCommitmentForwarder_Test is Testable, LenderCommitmentForwarder {
         Test.eq(
             super.getRequiredCollateral(
                 1e18, // 1 WETH loan
-                59e13, // 0.00059 WETH per USDC
+                59e13 * (1e18), // 0.00059 WETH per USDC base unit  //why does this work ?
```
Comment (passabilities) [inline packages/contracts/contracts/tests/LenderCommitmentForwarder_Test.sol]:
```suggestion
                59e13 * (1e18), // 0.00059 WETH per USDC base unit

File: packages/contracts/contracts/LenderCommitmentForwarder.sol (hunk 2 above)
Comment (passabilities) [reply packages/contracts/contracts/LenderCommitmentForwarder.sol]: whats not true about this?

File: packages/contracts/contracts/LenderCommitmentForwarder.sol [hunk 9]
```diff
@@ -361,31 +348,49 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
...
+     * @param _principalAmount The amount of currency to borrow for the loan.
+     * @param _maxPrincipalPerCollateralAmount The ratio for the amount of principal that can be borrowed for each amount of collateral.  This is expanded additionally by the principal decimals.
+     * @param _collateralTokenType The type of collateral for the loan either ERC20, ERC721, ERC1155, or None.
+     * @param _collateralTokenAddress The contract address for the collateral for the loan.
+     * @param _principalTokenAddress The contract address for the principal for the loan.
+     */
     function getRequiredCollateral(
         uint256 _principalAmount,
-        uint256 _maxPrincipalPerCollateralAmount,
+        uint256 _maxPrincipalPerCollateralAmount, //should be expressed expanded by collateralTokenDecimals and principalTokenDecimals
```
Comment (passabilities) [inline packages/contracts/contracts/LenderCommitmentForwarder.sol]:
```suggestion
        uint256 _maxPrincipalPerCollateralAmount,
```

**Review by passabilities — APPROVED** [review]

**Pull request merged**